ckanext/s3filestore/uploader.py

```diff
@@ -376,7 +376,8 @@ def update_data_dict(self, data_dict, url_field, file_field, clear_field):
 
         if not self.storage_path:
             return
-        if isinstance(self.upload_field_storage, ALLOWED_UPLOAD_TYPES):
+        if isinstance(self.upload_field_storage, ALLOWED_UPLOAD_TYPES) \
+                and self.upload_field_storage.filename:
```

Comment:
and unit test for this currently? if not please add at least one.

Reply:
There is now.